provider/pkg/provider/container.go

```diff
@@ -0,0 +1,327 @@
+			resource.PropertyMap{
+				"enabled": resource.NewBoolProperty(false),
+			}),
+		":setAutoscaling",
+		"POST"),
+	"management": updateNodePoolMapping(
+		mustParsePropertyPath("management"),
+		"management",
+		emptyPropVal,
+		":setManagement",
+		"POST"),
+	"upgradeSettings": updateNodePoolMapping(
+		mustParsePropertyPath("upgradeSettings"),
+		"upgradeSettings",
+		emptyPropVal,
+		"",
+		"PUT"),
+	"locations": updateNodePoolMapping(
+		mustParsePropertyPath("locations"),
+		"locations",
+		resource.NewArrayProperty([]resource.PropertyValue{}),
+		"",
+		"PUT"),
+	"version": updateNodePoolMapping(
+		mustParsePropertyPath("version"),
+		"nodeVersion",
+		resource.NewStringProperty(""),
+		"",
+		"PUT"),
+	"config.imageType":              updateNodePoolConfig(mustParsePropertyPath("config.imageType"), resource.NewStringProperty("")),
+	"config.kubeletConfig":          updateNodePoolConfig(mustParsePropertyPath("config.kubeletConfig"), emptyPropVal),
+	"config.linuxNodeConfig":        updateNodePoolConfig(mustParsePropertyPath("config.linuxNodeConfig"), emptyPropVal),
+	"config.workloadMetadataConfig": updateNodePoolConfig(mustParsePropertyPath("config.workloadMetadataConfig"), emptyPropVal),
+}
+
+var updateOrder = []string{"autoscaling", "config.imageType", "config.workloadMetadataConfig", "config.kubeletConfig",
+	"config.linuxNodeConfig", "initialNodeCount" /* is this the right attribute to update? */, "management",
+	"version", "locations", "upgradeSettings"}
+
+func mustParsePropertyPath(pattern string) resource.PropertyPath {
+	pp, err := resource.ParsePropertyPath(pattern)
+	contract.AssertNoError(err)
+	return pp
+}
+
+func updateNodepool(
+	providerInstance *googleCloudProvider,
+	urn resource.URN,
+	label string,
+	res *resources.CloudAPIResource,
+	inputs resource.PropertyMap,
+	oldState resource.PropertyMap,
+) (map[string]interface{}, error) {
+	// TODO wait for the cluster to be in resting state
+
+	// Extract old inputs from the `__inputs` field of the old state.
+	oldInputs := parseCheckpointObject(oldState)
+	outputs := resource.NewPropertyMapFromMap(oldState.Mappable())
+	if outputs.HasValue(resource.PropertyKey("__inputs")) {
+		delete(outputs, "__inputs")
+	}
+	// Delete the auto-injected __autonamed tag in old and new inputs to avoid spurious diffs.
+	if v, ok := oldInputs[autonamed]; ok && v.IsBool() {
+		delete(oldInputs, autonamed)
+	}
+
+	if v, ok := inputs[autonamed]; ok && v.IsBool() {
+		delete(inputs, autonamed)
+	}
+
+	diff := oldInputs.Diff(inputs)
+
+	var err error
+	if diff == nil {
+		logging.V(9).Infof("[%s] no diff found in update!", label)
+	} else {
+		// Calculate the detailed diff object containing information about replacements.
+		detailedDiff := calculateDetailedDiff(res, providerInstance.resourceMap.Types, diff)
+		logging.V(9).Infof("[%s] detailedDiff: %+v", label, detailedDiff)
+		logging.V(9).Infof("[%s] updateNodepool oldState: %+v", label, oldState)
+
+		for _, key := range updateOrder {
```

Comment:
Might want to log the update operation we're working on in each loop iteration

Reply:
Done